src/main/java/com/kodekonveyor/repo/backend/api/lookup/LookupTagByNameService.java

```diff
@@ -0,0 +1,54 @@
+package com.kodekonveyor.repo.backend.api.lookup;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.kodekonveyor.exception.ValidationException;
+import com.kodekonveyor.repo.api.LerpoiEntity;
+import com.kodekonveyor.repo.api.LerpoiEntityRepository;
+import com.kodekonveyor.repo.api.SumtiConstants;
+import com.kodekonveyor.repo.api.SumtiDTO;
+import com.kodekonveyor.repo.api.SumtiEntity;
+import com.kodekonveyor.repo.api.SumtiEntityRepository;
+
+@Service
+public class LookupTagByNameService {
+
+  @Autowired
+  private LerpoiEntityRepository lerpoiEntityRepository;
+
+  @Autowired
+  private SumtiEntityRepository sumtiEntityRepository;
+
+  public List<SumtiDTO> call(final String repoName) {
+    final Optional<LerpoiEntity> lerpoiEntity =
+        lerpoiEntityRepository.findByName(repoName);
+    if (lerpoiEntity.isEmpty())
+      throw new ValidationException(
+          SumtiConstants.LERPOI_NOT_FOUND_EXCEPTION_FOR_TAG_LOOKUP
+      );
+    final List<SumtiEntity> sumtiEntity =
+        sumtiEntityRepository.findByLerpoi(lerpoiEntity.get());
```

Comment:
You can use traversal query instead of 2 queries.
https://thorben-janssen.com/ultimate-guide-derived-queries-with-spring-data-jpa/#Traverse_associations_in_derived_queries
```sumtiEntityRepository.findByLerpoiName(name);```